libgalois/include/katana/PropertyGraph.h

```diff
@@ -155,23 +155,30 @@ class KATANA_EXPORT GraphTopology {
 /// comprise the physical representation of the logical property graph.
 class KATANA_EXPORT PropertyGraph {
 public:
-  /// TypeSetID uniquely identifies/contains a combination/set of types
-  /// TypeSetID is represented using 8 bits
-  /// TypeSetID for nodes is distinct from TypeSetID for edges
-  using TypeSetID = uint8_t;
-  static constexpr TypeSetID kUnknownType = TypeSetID{0};
-  static constexpr TypeSetID kInvalidType =
-      std::numeric_limits<TypeSetID>::max();
-  /// A set of TypeSetIDs
-  using SetOfTypeSetIDs =
-      std::bitset<std::numeric_limits<TypeSetID>::max() + 1>;
-  /// A set of type names
+  /// EntityTypeID uniquely identifies an entity (node or edge) type
+  /// EntityTypeID for nodes is distinct from EntityTypeID for edges
+  /// This type may either be an atomic type or an intersection of atomic types
+  /// EntityTypeID is represented using 8 bits
+  using EntityTypeID = uint8_t;
+  static constexpr EntityTypeID kUnknownType = EntityTypeID{0};
```

Comment:
Does `kUnknownType` mean that the value could have any type? If so then it's the union of all atomic types. This is usually called `Top`.

Reply:
No, it means the type is unspecified/unknown. It cannot be any type.

Reply:
If we do not know the  type of some entity, we can still say that it must have SOME type. And if it has some type, then it is the member of the union of all types. And that union is called "top" or sometimes "any".

Reply:
No, it is not equivalent to "any" type. If a user requests nodes with a particular type, this node with unknown type will not be returned. It is a special type. We can enforce that all nodes/edges/entities must have types but we are not planning to do that (at least in the near future).

Reply:
So you view Unknown as it's own type which is a subtype of Any and without any relationship to any other type? That's reasonable I guess. In most systems, you wouldn't have a name for this separate from Any. Instead, values which have the most specific type Any are implicitly unknown because you cannot determine a more specific type for them.

The only issue I can see with having an Unknown type is that since types are not disjoint, having type Unknown doesn't preclude having other types. Encoding unknown by having Any as the most specific type of a value avoids this issue since having any other type implicitly means we know something about the value so Any isn't the most specific type any more.